pyproject.toml

```diff
@@ -86,6 +86,7 @@ snowflake-connector-python = { version = ">=2.7.10,<4", optional = true }
 snowflake-sqlalchemy = { version = ">=1.4.1,<2", optional = true }
 sqlalchemy = { version = ">=1.4,<3", optional = true }
 sqlalchemy-views = { version = ">=0.3.1,<1", optional = true }
+streamlit = { version = ">=1.22,<2", optional = true, python = ">=3.8,!=3.9.7,<4.0" }
```

Comment:
Maybe we should put this in dev deps only?

I'm not sure what benefit there is to including it as a proper optional dependency since in the case that someone is using our `ibis.streamlit` connection implementation `streamlit` must be installed.

Reply:
Split it out to an dev/test-only dependency.

Reply:
Makes sense to me. Nice to keep streamlit and deps out of our already large dev environment too.